### PR TITLE
arguments.caller test improvements

### LINKED ISSUE #952: arguments.caller and arguments.callee should be a TypeError in ES5 but should not in ES2017
So should we remove this test from ES5 tab? Look at Chrome 56 at [ES5 tab](http://kangax.github.io/compat-table/es5/) `Strict mode` tests group and [ES2016+ tab](http://kangax.github.io/compat-table/es2016plus/) `2017 misc` group

### PR DESCRIPTION
**Pull request by chicoxyzzy**
Fixes #952

### CODE REVIEW

File: data-es5.js [hunk 1]
```diff
@@ -1626,10 +1621,12 @@ exports.tests = [
     res: temp.strict,
   },
   {
-    name: 'arguments.caller and arguments.callee is a TypeError',
+    name: 'arguments.caller removed or is a TypeError and arguments.callee is a TypeError',
     exec: function() {/*
       'use strict';
-      try { arguments.caller; return false; } catch (err) { if (!(err instanceof TypeError)) return false; }
+      if (Object.getOwnPropertyDescriptor(arguments,'caller')) {
```
Comment (ljharb) [inline data-es5.js]: we should probably split up `caller` and `callee` into separate tests?

Comment (chicoxyzzy) [reply data-es5.js]: done

File: data-es5.js [hunk 2]
```diff
@@ -1626,10 +1621,33 @@ exports.tests = [
     res: temp.strict,
   },
   {
-    name: 'arguments.caller and arguments.callee is a TypeError',
+    name: 'arguments.caller removed or is a TypeError',
+    exec: function() {/*
+      'use strict';
+      if (Object.getOwnPropertyDescriptor(arguments,'caller')) {
```
Comment (ljharb) [inline data-es5.js]: `getOwnPropertyDescriptor` doesn't exist in ES3 - i think you need to use `in` here.

Comment (chicoxyzzy) [reply data-es5.js]: right. fixed

**Review by ljharb — APPROVED** [review]
LGTM, assuming it's been tested in browsers pre-poison-pill, during-poison-pill, and post-poison-pill.

**Pull request merged**